src/ser/mod.rs

```diff
@@ -176,16 +169,34 @@ impl Serializer {
         }
     }
 
+    fn indentation_enabled(&self) -> bool {
+        // TODO(torkleyy): consider changing this to <=
+        // TODO: (see https://github.com/ron-rs/ron/issues/174)
+        self.pretty.as_ref().map(|(config, pretty)| pretty.indent < config.depth_limit)
+            .unwrap_or(false)
+    }
+
+    fn pop_if_comma(&mut self) {
+        if self.output.as_bytes().last() == Some(&b',') {
+            self.output.pop();
+        }
+    }
+
     fn indent(&mut self) {
         if let Some((ref config, ref pretty)) = self.pretty {
-            if pretty.indent < config.depth_limit {
+            if self.indentation_enabled() {
                 self.output
                     .extend((0..pretty.indent).map(|_| config.indentor.as_str()));
             }
         }
     }
 
     fn end_indent(&mut self) {
+        // Remove trailing comma if no indentation is used
+        if !self.indentation_enabled() {
+            self.pop_if_comma();
```

Comment:
could there be a better way to check this? i.e. pop if there was at least one item in the current block, instead of checking the output stream

Reply:
Yes, let's do that.